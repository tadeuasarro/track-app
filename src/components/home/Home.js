import Footer from '../footer/Footer';
import './home.css';

const Home = () => (
  <div>
    <div className="home-container">
      <div className="home-summary">
        <div className="home-summary-container">
          <div className="home-summary-info">1.0 K</div>
          <p>Expent</p>
        </div>
        <div className="home-summary-container">
          <div className="home-summary-info">2.0 K</div>
          <p>Target</p>
        </div>
        <div className="home-summary-container">
          <div className="home-summary-info">1.0 K</div>
          <p>Balance</p>
        </div>
      </div>
      <div className="home-catalogue">
        this is some content
      </div>
    </div>
    <Footer />
  </div>
);

export default Home;
