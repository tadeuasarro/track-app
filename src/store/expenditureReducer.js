import { SET_EXPENDITURES } from '../actions/expenditure';
import filterExpenditures from '../helpers/filterExpenditures';

const initialState = {
  expenditures: [],
  summary: false,
};

const expenditureReducer = (state = initialState, action) => {
  switch (action.type) {
    case SET_EXPENDITURES:
      return {
        expenditures: action.payload,
        summary: filterExpenditures(action.payload),
      };
    default:
      return state;
  }
};

export default expenditureReducer;
