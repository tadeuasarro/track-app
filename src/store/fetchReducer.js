import { SET_LOADING, SET_ERRORS } from '../actions/fetch';

const initialState = {
  loading: false,
  errors: false,
};

const fetchReducer = (state = initialState, action) => {
  switch (action.type) {
    case SET_LOADING:
      return {
        loading: true,
        errors: false,
      };
    case SET_ERRORS:
      return {
        loading: false,
        errors: action.payload,
      };
    default:
      return state;
  }
};

export default fetchReducer;
