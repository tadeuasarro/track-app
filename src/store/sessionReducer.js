import { SET_CURRENT_USER, UPDATE_USER_TARGET } from '../actions/session';

const initialState = {
  username: false,
  id: false,
  target: false,
  token: false,
};

const sessionReducer = (state = initialState, action) => {
  switch (action.type) {
    case SET_CURRENT_USER:
      return {
        username: action.payload.user.username,
        id: action.payload.user.id,
        target: action.payload.user.target,
        token: action.payload.token,
      };
    case UPDATE_USER_TARGET:
      return {
        ...state,
        target: action.payload,
      };
    default:
      return state;
  }
};

export default sessionReducer;
