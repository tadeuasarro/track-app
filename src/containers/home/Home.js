import { useSelector } from 'react-redux';
import Footer from '../../components/footer/Footer';
import Loading from '../../components/loading/Loading';
import './home.css';

const Home = () => {
  const { session, expenditure } = useSelector(state => state);

  const { target } = session;

  if (!expenditure.expenditures) return <Loading />;

  const {
    total,
    education,
    grocery,
    health,
    leisure,
    living,
    transport,
  } = expenditure.summary;

  return (
    <div>
      <div className="home-container">
        <div className="home-summary">
          <div className="home-summary-container">
            <div className="home-summary-info">
              {(total / 1000).toFixed(1)}
              &nbsp;
              k
            </div>
            <p>Expent</p>
          </div>
          <div className="home-summary-container">
            <div className="home-summary-info">
              {(target / 1000).toFixed(1)}
              &nbsp;
              k
            </div>
            <p>Target</p>
          </div>
          <div className="home-summary-container">
            <div className="home-summary-info">
              {((target - total) / 1000).toFixed(1)}
              &nbsp;
              k
            </div>
            <p>Balance</p>
          </div>
        </div>
        <div className="home-catalogue-container">
          <div className="home-catalogue">
            <div className="home-catalogue-item">
              <i className="fas fa-graduation-cap" />
              <div>
                <p>
                  {(education / 1000).toFixed(1)}
                  &nbsp;
                  k
                </p>
                <span>Education</span>
              </div>
            </div>
            <div className="home-catalogue-item">
              <i className="fas fa-shopping-cart" />
              <div>
                <p>
                  {(grocery / 1000).toFixed(1)}
                  &nbsp;
                  k
                </p>
                <span>Grocery</span>
              </div>
            </div>
            <div className="home-catalogue-item">
              <i className="fas fa-heartbeat" />
              <div>
                <p>
                  {(health / 1000).toFixed(1)}
                  &nbsp;
                  k
                </p>
                <p>Health</p>
              </div>
            </div>
            <div className="home-catalogue-item">
              <i className="fas fa-gamepad" />
              <div>
                <p>
                  {(leisure / 1000).toFixed(1)}
                  &nbsp;
                  k
                </p>
                <span>Leisure</span>
              </div>
            </div>
            <div className="home-catalogue-item">
              <i className="fas fa-home" />
              <div>
                <p>
                  {(living / 1000).toFixed(1)}
                  &nbsp;
                  k
                </p>
                <span>Living</span>
              </div>
            </div>
            <div className="home-catalogue-item">
              <i className="fas fa-bus" />
              <div>
                <p>
                  {(transport / 1000).toFixed(1)}
                  &nbsp;
                  k
                </p>
                <span>Transport</span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <Footer />
    </div>
  );
};

export default Home;
