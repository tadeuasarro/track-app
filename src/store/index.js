import { combineReducers, createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import sessionReducer from './sessionReducer';
import expenditureReducer from './expenditureReducer';
import fetchReducer from './fetchReducer';

const store = createStore(
  combineReducers({
    session: sessionReducer,
    expenditure: expenditureReducer,
    fetch: fetchReducer,
  }),
  applyMiddleware(thunk),
);

export default store;
