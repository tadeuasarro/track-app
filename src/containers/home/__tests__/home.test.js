import { render, screen, act } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../../store/index';
import Home from '../Home';

describe('Rendering component', () => {
  it('creates a Home component', () => {
    act(() => {
      render(
        <BrowserRouter>
          <Provider store={store}>
            <Home />
          </Provider>
          ,
        </BrowserRouter>,
      );
    });
    // eslint-disable-next-line
    const element = screen.getByText('Target');
  });
});

describe('Display', () => {
  it('renders correctly', () => {
    const comp = renderer.create(
      <BrowserRouter>
        <Provider store={store}>
          <Home />
        </Provider>
        ,
      </BrowserRouter>,
    ).toJSON();
    expect(comp).toMatchSnapshot();
  });
});
