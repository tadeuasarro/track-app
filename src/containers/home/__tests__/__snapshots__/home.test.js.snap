// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display renders correctly 1`] = `
Array [
  <div>
    <div
      className="home-container"
    >
      <div
        className="home-summary"
      >
        <div
          className="home-summary-container"
        >
          <div
            className="home-summary-info"
          >
            NaN
              k
          </div>
          <p>
            Expent
          </p>
        </div>
        <div
          className="home-summary-container"
        >
          <div
            className="home-summary-info"
          >
            0.0
              k
          </div>
          <p>
            Target
          </p>
        </div>
        <div
          className="home-summary-container"
        >
          <div
            className="home-summary-info"
          >
            NaN
              k
          </div>
          <p>
            Balance
          </p>
        </div>
      </div>
      <div
        className="home-catalogue-container"
      >
        <div
          className="home-catalogue"
        >
          <div
            className="home-catalogue-item"
          >
            <i
              className="fas fa-graduation-cap"
            />
            <div>
              <p>
                NaN
                  k
              </p>
              <span>
                Education
              </span>
            </div>
          </div>
          <div
            className="home-catalogue-item"
          >
            <i
              className="fas fa-shopping-cart"
            />
            <div>
              <p>
                NaN
                  k
              </p>
              <span>
                Grocery
              </span>
            </div>
          </div>
          <div
            className="home-catalogue-item"
          >
            <i
              className="fas fa-heartbeat"
            />
            <div>
              <p>
                NaN
                  k
              </p>
              <p>
                Health
              </p>
            </div>
          </div>
          <div
            className="home-catalogue-item"
          >
            <i
              className="fas fa-gamepad"
            />
            <div>
              <p>
                NaN
                  k
              </p>
              <span>
                Leisure
              </span>
            </div>
          </div>
          <div
            className="home-catalogue-item"
          >
            <i
              className="fas fa-home"
            />
            <div>
              <p>
                NaN
                  k
              </p>
              <span>
                Living
              </span>
            </div>
          </div>
          <div
            className="home-catalogue-item"
          >
            <i
              className="fas fa-bus"
            />
            <div>
              <p>
                NaN
                  k
              </p>
              <span>
                Transport
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="footer-container"
    >
      <a
        className="footer-link"
        href="/measures"
        onClick={[Function]}
      >
        <i
          className="fas fa-balance-scale-right"
        />
      </a>
      <a
        className="footer-link"
        href="/"
        onClick={[Function]}
      >
        <i
          className="fas fa-bars"
        />
      </a>
      <a
        className="footer-link"
        href="/progress"
        onClick={[Function]}
      >
        <i
          className="fas fa-tasks"
        />
      </a>
      <a
        className="footer-link"
        href="/more"
        onClick={[Function]}
      >
        <i
          className="fas fa-info"
        />
      </a>
    </div>
  </div>,
  ",",
]
`;
